### PR TITLE
Feature/signin guest mentors

### PR DESCRIPTION
**Pull request by Lytigas**

### CODE REVIEW

File: registration_2019/dayof.py [hunk 1]
```diff
@@ -62,22 +81,26 @@ class SignInEndpoint(Resource):
...
         self.parser.add_argument('user_id',            type=strn, required=True)
-        self.parser.add_argument('badge_data',         type=strn, required=True)
+        self.parser.add_argument('badge_data',         type=badge_data, required=True)
 
     @auth
     def post(self):
         args = self.parser.parse_args()
         return sign_in(args['user_id'], args['badge_data'])
 
     def get(self):
```
Comment (nayaverdier) [inline registration_2019/dayof.py]: Should be auth'd

File: registration_2019/dayof.py [hunk 2]
```diff
@@ -6,27 +6,46 @@
...
-        return {"message": "User already signed in"}, 400
-    sign_in = SignIn(badge_data=badge_data)
-    db.session.add(sign_in)
-    # Could use the modify method, but interaction would be non-trivial given that its designed for forward facing API
-    signup.sign_in = sign_in
-    db.session.commit()
-
-    return {"status": "ok"}
+    if signup:
+        if signup.sign_in:
```
Comment (nayaverdier) [inline registration_2019/dayof.py]: If the SignIn has `signed_out` true, this will return that they have already signed in?

Comment (Lytigas) [reply registration_2019/dayof.py]: I interpreted sign in and sign out as a one time thing, if we wanted to add tracking whether they are currently at the event then I would need to add more code.

File: registration_2019/dayof.py (hunk 1 above)
Comment (Lytigas) [reply registration_2019/dayof.py]: Not sure what you're referring to, if its the GET method, I don't think it needs to be as it only shows counts. The Post method is authed.

**Pull request merged**